src/helpers/index.ts

```diff
@@ -5,12 +5,40 @@ import type {
 
 const GWEI = 10 ** 9;
 
+export async function recoverWithToken(
+  token: string,
+  params: Parameters<typeof window.commands.recover>
+) {
+  const [coin, rest] = params;
+  console.log(rest);
+  try {
+    return await window.commands.recover(token, rest);
+  } catch (e) {
+    return await window.commands.recover(coin, rest);
+  }
+}
+
+export async function getChainWithToken(
```

Comment:
better for it to be called `getTokenChain` since you are getting the chain that the token belongs too